### ISSUE TITLE
Bold Text settings are not working for embedded webview

### ISSUE THREAD

**Issue by hannah-hyj** [flutter/flutter#178315]
internal issues reported b/458517170

Internal issues that app (using webview) on both iOS and Android fails to display text in bold when the accessibility bold text setting is enabled.

**Comment by loic-sharma** [comment]
I believe Flutter Web cannot automatically detect that it should bold text on iOS. Safari doesn't automatically bold the content of websites if you turn on iOS's `Bold text` accessibility feature.

The iOS app that embeds Flutter Web as a web view will need to wire this in manually.

<details>
<summary>How to manually enable bold text...</summary>

Here's a rough sketch of what this might look like:

1. Make the iOS app's native code listens to [`UIAccessibility.boldTextStatusDidChangeNotification`](https://developer.apple.com/documentation/uikit/uiaccessibility/boldtextstatusdidchangenotification)
2. When the notification is posted, the iOS app's native code looks up whether text is bolded using [`UIAccessibility.isBoldTextEnabled`](https://developer.apple.com/documentation/uikit/uiaccessibility/isboldtextenabled).
3. The iOS app sends a message to its web view to notify it of the latest bold text status. For example, it can use [`WKWebView.evaluateJavaScript`](https://developer.apple.com/documentation/webkit/wkwebview/evaluatejavascript(_:completionhandler:))
4. In Javascript, use [JS Interop](https://dart.dev/interop/js-interop/usage) to notify the Flutter Web app of the latest bold text status.
5. In your app, use a `MediaQuery` widget to set [`MediaQueryData.boldText`](https://api.flutter.dev/flutter/widgets/MediaQueryData/boldText.html) to the latest value. Something like:

   ```dart
   class BoldTextOverride extends StatelessWidget {
     const BoldTextOverride({
       super.key,
       required this.boldText,
       required this.child,
     });

     final bool? boldText;
     final Widget child;

     @override
     Widget build(BuildContext context) {
       if (boldText == null) {
         return child;
       }

       return MediaQuery(
         data: MediaQuery.of(context).copyWith(boldText: boldText),
         child: child,
       );
     }
   }
   ```

   You'll want this widget to be high in the widget tree, as a parent of all text widgets that should be bolded.

</details>

Since this isn't something Flutter can support automatically, I believe we should close this issue. Please let me know if you have any questions or concerns!
